### PR TITLE
Revert "Add string-split primitive"

### PR DESCRIPTION
**Pull request by tymm**
@kjmcqueen Are you handling the `NARGS == 3` case correctly? I can't see that you return the equivalent of `PRIM_ERROR("Fixnum expected for limit")`.

### CODE REVIEW

File: roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala [hunk 1]
```diff
@@ -244,81 +242,6 @@ object rblstring {
...
-    override def fn(ctxt: Ctxt): Either[PrimError, Tuple] = {
-      val elem = ctxt.argvec.elem
-
-      for {
-        str <- checkType[RblString](0, elem) // Ensure arg0 is a RblString
-        sep <- checkType[RblString](1, elem) // Ensure arg1 is a RblString
-        w <- checkType[Fixnum](2, elem) // Ensure optional arg2 is a Fixnum
-            match {
-              case Right(v)                 => Right(v)
-              case Left(TypeMismatch(_, _)) => Left(TypeMismatch(2, Fixnum.getClass.getName))
```
Comment (kjmcqueen) [inline roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala]: This line should be doing that.  Is that not true?

File: roscala/src/test/scala/coop/rchain/rosette/prim/RblStringSpec.scala [hunk 2]
```diff
@@ -805,112 +805,4 @@ class RblStringSpec extends FlatSpec with Matchers {
...
-
-  it should "fail for invalid Count type" in {
-    val sep = ".,-=/*"
-    val str = "aZ,bY.cX-dW=eV/fU*gT"
-    val res = Tuple(Seq(RblString("aZ"), RblString("bY"), RblString("cX-dW=eV/fU*gT")))
-
-    val parms   = Tuple(Seq(RblString(str), RblString(sep), RblString("foo")))
-    val newCtxt = ctxt.copy(nargs = 3, argvec = parms)
-    stringSplit.fn(newCtxt) should be('left)
-  }
```
Comment (kjmcqueen) [inline roscala/src/test/scala/coop/rchain/rosette/prim/RblStringSpec.scala]: This test is for that very case.

File: roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala [hunk 3]
```diff
@@ -244,81 +242,6 @@ object rblstring {
...
-              case Left(_)                  => Right(Fixnum(Int.MaxValue))
```
Comment (kjmcqueen) [inline roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala]: This line handles the optional count case.

Comment (tymm) [reply roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala]: I guess this is the line that I don't understand yet. Can you tell me Rosette returns `Fixnum(Int.MaxValue)`?

Comment (tymm) [reply roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala]: Oh, my bad: `int lim = FIXVAL(MAX_FIXNUM)`.

Comment (kjmcqueen) [reply roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala]: RBLstring.cc line 466

    int lim = FIXVAL(MAX_FIXNUM);
    if (NARGS == 3) {
        if (IS_FIXNUM(ARG(2))) {
            lim = FIXVAL(ARG(2));
        } else {
            return PRIM_ERROR("Fixnum expected for limit");
        }
    }

**Pull request closed without merging**